### ISSUE TITLE
[BUG] Got "TypeError: not a string" when tried to run predictor.fit() with custom 'model.hf_text.checkpoint_name' hyperparameters 

### ISSUE THREAD

**Issue by muazhari** [autogluon/autogluon#2449]
- [x] I have checked that this bug exists on the latest stable version of AutoGluon
- [x] and/or I have checked that this bug exists on the latest mainline of AutoGluon via source installation

**Describe the bug**
I got an error while running a rating prediction task on the e-commerce review dataset. I am using the latest stable version of AutoGluon. Error message: `TypeError: not a string`

**Expected behavior**
I expect these `predictor.fit()` method works normally 
```
predictor.fit(
    train_data,
    hyperparameters={
        'model.hf_text.checkpoint_name': 'indobenchmark/indobert-lite-large-p2'
    }
)
``` 

**To Reproduce**
Use this provided Kaggle notebook: 
https://www.kaggle.com/code/muazhari/e-commerce-review-rating-prediction-error/notebook

**Installed Versions**
from pip show details:
```
Name: autogluon
Version: 0.6.0
Summary: AutoML for Image, Text, and Tabular Data
Home-page: https://github.com/awslabs/autogluon
Author: AutoGluon Community
Author-email: 
License: Apache-2.0
Location: /opt/conda/lib/python3.7/site-packages
Requires: autogluon.core, autogluon.features, autogluon.multimodal, autogluon.tabular, autogluon.text, autogluon.timeseries, autogluon.vision
```
from autogluon show_versions details:
```python
INSTALLED VERSIONS
------------------
date                   : 2022-11-20
time                   : 18:23:33.730980
python                 : 3.7.12.final.0
OS                     : Linux
OS-release             : 5.15.65+
Version                : #1 SMP Thu Nov 10 10:13:28 UTC 2022
machine                : x86_64
processor              : x86_64
num_cores              : 2
cpu_ram_mb             : 16006
cuda version           : 11.470.82.01
num_gpus               : 1
gpu_ram_mb             : [15543]
avail_disk_size_mb     : 16111

accelerate             : 0.12.0
albumentations         : 1.1.0
autogluon.common       : 0.6.0
autogluon.core         : 0.6.0
autogluon.features     : 0.6.0
autogluon.multimodal   : 0.6.0
autogluon.tabular      : 0.6.0
autogluon.text         : 0.6.0
autogluon.timeseries   : 0.6.0
autogluon.vision       : 0.6.0
boto3                  : 1.24.93
catboost               : 1.1
dask                   : 2021.11.2
defusedxml             : 0.7.1
distributed            : 2021.11.2
evaluate               : 0.2.2
fairscale              : 0.4.6
fastai                 : 2.7.9
gluoncv                : 0.11.0
gluonts                : 0.11.1
hyperopt               : 0.2.7
joblib                 : 1.2.0
jsonschema             : 4.6.1
lightgbm               : 3.3.2
matplotlib             : 3.5.3
networkx               : 2.5
nlpaug                 : 1.1.10
nltk                   : 3.7
nptyping               : 1.4.4
numpy                  : 1.21.6
omegaconf              : 2.1.2
openmim                : None
pandas                 : 1.3.5
PIL                    : 9.0.1
psutil                 : 5.8.0
pycocotools            : None
pytorch-metric-learning: None
pytorch_lightning      : 1.7.7
ray                    : 2.0.0
requests               : 2.28.1
scipy                  : 1.7.3
sentencepiece          : 0.1.97
seqeval                : None
setuptools             : 59.8.0
skimage                : 0.19.3
sklearn                : 1.0.2
smart_open             : 5.2.1
statsmodels            : 0.13.2
text-unidecode         : None
timm                   : 0.6.11
torch                  : 1.11.0
torchmetrics           : 0.8.2
torchtext              : 0.12.0
torchvision            : 0.12.0
tqdm                   : 4.64.0
transformers           : 4.23.1
xgboost                : 1.6.2
```

**Comment by FANGAreNotGnu** [comment]
This is because sentencepiece is not used in this model. Here is a minimal code example to reproduce the error:

```
import pandas as pd
from autogluon.multimodal import MultiModalPredictor

train_data = pd.DataFrame({"text": ["high", "higher", "low"], "rating": [2,3,0]})
target_column = "rating"

predictor = MultiModalPredictor(
    label=target_column
)

predictor.fit(
    train_data,
    hyperparameters={
        'env.num_gpus': 1,
        'model.hf_text.checkpoint_name': 'indobenchmark/indobert-lite-large-p2',
    }
)
```

And here is a fix: #2550, where we switch to try BertTokenizer if such error occurs.

**Comment by sxjscience** [comment]
@muazhari The issue should have been fixed. Feel free to reopen it if you still have problems.
